frontend/ui/widget/skimtowidget.lua

```diff
@@ -0,0 +1,278 @@
+local InputContainer = require("ui/widget/container/inputcontainer")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local FrameContainer = require("ui/widget/container/framecontainer")
+local CenterContainer = require("ui/widget/container/centercontainer")
+local VerticalGroup = require("ui/widget/verticalgroup")
+local OverlapGroup = require("ui/widget/overlapgroup")
+local CloseButton = require("ui/widget/closebutton")
+local TextWidget = require("ui/widget/textwidget")
+local LineWidget = require("ui/widget/linewidget")
+local GestureRange = require("ui/gesturerange")
+local Button = require("ui/widget/button")
+local UIManager = require("ui/uimanager")
+local Screen = require("device").screen
+local Device = require("device")
+local Geom = require("ui/geometry")
+local Font = require("ui/font")
+local _ = require("gettext")
+local Blitbuffer = require("ffi/blitbuffer")
+local HorizontalGroup = require("ui/widget/horizontalgroup")
+local ProgressWidget = require("ui/widget/progresswidget")
+local VerticalSpan = require("ui/widget/verticalspan")
+
+local SkimToWidget = InputContainer:new{
+    title_face = Font:getFace("tfont", 22),
+    width = nil,
+    height = nil,
+}
+
+function SkimToWidget:init()
+    self.medium_font_face = Font:getFace("ffont", 20)
+    self.screen_width = Screen:getSize().w
+    self.screen_height = Screen:getSize().h
+    self.span = math.ceil(self.screen_height * 0.01)
+    self.width = self.screen_width * 0.95
+    if Device:hasKeys() then
+        self.key_events = {
+            Close = { {"Back"}, doc = "close skimto page" }
+        }
+    end
+    if Device:isTouchDevice() then
+        self.ges_events = {
+            TapCloseGT = {
+                GestureRange:new{
+                    ges = "tap",
+                    range = Geom:new{
+                        x = 0, y = 0,
+                        w = self.screen_width,
+                        h = self.screen_height,
+                    }
+                },
+            },
+         }
+    end
+    local curr_page
+    if self.document.info.has_pages then
+        self.dialog_title = _("Go to Page")
+        curr_page = self.ui.paging.current_page
```

Comment:
By using `self.ui`, this widget is already coupled with reader app, so it's better to move it into `apps/reader`. You can even move the code into `readergoto.lua`.